lib/components/popups/course-search/Cart.tsx

```diff
@@ -33,33 +29,34 @@ const Cart: FC<{}> = () => {
 
   // FOR DUMMY FILTER TESTING TODO REMOVE
   // TODO : double check the initial state on this hook. do i even need this if stored in redux?
-  const [selectedRequirement, setSelectedRequirement] =
-    useState<requirements>(emptyRequirements);
   const [cartFilter, setCartFilter] = useState<string>('');
   const [isCartFilterLoaded, setIsCartFilterLoaded] = useState<boolean>(false);
   const [textFilterInputValue, setTextFilterInputValue] = useState<string>('');
 
   // Redux selectors and dispatch
   const dispatch = useDispatch();
-  const distrs = useSelector(selectSelectedDistribution);
-  const currentPlanCourses = useSelector(selectCurrentPlanCourses);
-  const cartAdd = useSelector(selectCartAdd);
+  const dist = useSelector(selectSelectedDistribution);
+  const fine = useSelector(selectSelectedFineReq);
   const pageIndex = useSelector(selectPageIndex);
 
+  // upda
   useEffect(() => {
-    setCartFilter(selectedRequirement.expr);
-    dispatch(updatePageIndex(0));
+    if (dist) {
+      setCartFilter(dist.criteria);
+      setIsCartFilterLoaded(true);
+      dispatch(updatePageIndex(0));
+    }
```

Comment:
in cart we listen to changes in selectedDistribution and selectedFineRequirement to update search filter